pac4j-http/src/main/java/org/pac4j/http/client/direct/HeaderClient.java

```diff
@@ -52,10 +52,12 @@ public HeaderClient(final String headerName, final String prefixHeader,
 
     @Override
     protected void clientInit(final WebContext context) {
-        CommonHelper.assertNotBlank("headerName", this.headerName);
-        CommonHelper.assertNotNull("prefixHeader", this.prefixHeader);
+        if (getCredentialsExtractor() == null) {
```

Comment:
In fact, you don't need that test as the `defaultCredentialsExtractor` already does that...